db/import.go

```diff
@@ -218,12 +218,21 @@ func (db *Database) importDoc(docid string, body Body, isDelete bool, existingDo
 
 		// If there's a filter function defined, evaluate to determine whether we should import this doc
 		if db.DatabaseContext.Options.ImportOptions.ImportFilter != nil {
-			shouldImport, err := db.DatabaseContext.Options.ImportOptions.ImportFilter.EvaluateFunction(body)
-			if err != nil {
+			var shouldImport bool
+			var importErr error
+
+			if isDelete {
```

Comment:
As discussed, there are three scenarios we need to cover:
- isDelete=true, body=nil : create a new deleteBody (already handled here)
- isDelete=false : use body (already handled here)
- isDelete=true, body != nil: take a shallow copy of body, set BodyDeleted:true (needs handling)

Reply:
Make sense, committed changes to take a shallow copy of the body and set _deleted property the import is for delete and the body is not empt.